pkg/util/hostname/providers.go

```diff
@@ -5,15 +5,45 @@
 
 package hostname
 
-import "context"
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util/hostname/validate"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
 
 // Provider is a generic function to grab the hostname and return it
 type Provider func(ctx context.Context, options map[string]interface{}) (string, error)
 
-// ProviderCatalog holds all the various kinds of hostname providers
-var ProviderCatalog = make(map[string]Provider)
+// providerCatalog holds all the various kinds of hostname providers
+var providerCatalog = make(map[string]Provider)
 
 // RegisterHostnameProvider registers a hostname provider as part of the catalog
 func RegisterHostnameProvider(name string, p Provider) {
-	ProviderCatalog[name] = p
+	providerCatalog[name] = p
+}
+
+// GetProvider returns a Provider if it was register before.
+func GetProvider(providerName string) Provider {
+	if provider, found := providerCatalog[providerName]; found {
+		return provider
+	}
+	return nil
+}
+
+// GetHostname returns the hostname for a specific Provider if it was register
+func GetHostname(ctx context.Context, providerName string, options map[string]interface{}) (string, error) {
+	if provider, found := providerCatalog[providerName]; found {
+		log.Debugf("GetHostname trying %s through API server...", providerName)
```

Comment:
This message may need a bit tweaking as we won't always go through an API server (e.g. `file` provider)